src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebResponseHelper.CoreClr.cs

```diff
@@ -16,6 +18,24 @@ internal static string GetCharacterSet(HttpResponseMessage response)
             string characterSet = response.Content.Headers.ContentType.CharSet;
             return characterSet;
         }
+        
+        internal static Dictionary<string, IEnumerable<string>> GetHeadersDictionary(HttpResponseMessage response)
+        {
+            var headers = new Dictionary<string, IEnumerable<string>>(StringComparer.OrdinalIgnoreCase);
+            foreach (var entry in response.Headers)
+            {
+                headers[entry.Key] = entry.Value;
+            }
+            if (response.Content != null)
+            {
+                foreach (var entry in response.Content.Headers)
+                {
+                    headers[entry.Key] = entry.Value;
```

Comment:
Interesting, can there be duplicate headers in `response.Headers` and `response.Content.Headers` ? Have second ones a high ptiority?

Reply:
No. The headers in `Response.Content.Headers` will never exist in `response.Headers`. #4494 added this to address the lack of `Content-Type` in the `WebResponseObject.Headers`

Reply:
Please clarify - my understanding from #4494 is that `headers` already contains `response.Content.Headers`, correct?

Reply:
That was add in #4494. Previous to that, `WebResponseObject.Headers` did not contain any of the headers in `HttpResponseMessage.Content.Headers`. CoreFX has split the response headers so that those related to content are always in `HttpResponseMessage.Content.Headers` and the rest of the response headers are in `HttpResponseMessage.Headers`. 

To clarify, I'm not adding or changing any of the logic in this PR for the creation of the Dictionary. Just moving it so that a new dictionary is not created on every Get to `Headers` and to make the logic available outside of `WebResponseObject`. The existing logic is there on purpose and we have existing tests to ensure it.

Reply:
Thanks for clarify.
So `foreach (var entry in response.Content.Headers) ` is duplicate code and could we remove it?
Sorry for late question - if CoreFX split the headers why we join them again? Only for backward compatibility?

Reply:
We are joining them for backwards compat. In 5.1 and earlier all headers are in `WebResponseObject.Headers`. In 6.0 the content headers are buried in `WebResponseObject.BaseResponse.Content.Headers` unless we promote them to the `WebResponseObject.Headers` dictionary. One would reasonably expect the `Headers` property to contain all headers.

Reply:
@markekraus can you please add a comment to the code to clarify that there won't be duplicate headers in `response.Headers` and `response.Content.Headers`? It would be very helpful to other people who look at the code later.

Reply:
@daxian-dbw should I also include a note about the content headers being added to the headers dictionary for backwards compatibility? or is the note about the distinct headers sufficient?

Reply:
I think it would also be helpful to mention the backward compatibility issue. Thanks @markekraus!

Reply:
Comments added.